client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,184 @@
+  import { onMount } from "svelte";
+  import { lobbyReturn, getLobbyRole } from "$lib/firebase/firestore-functions";
+  import type { ErrorResponse, LobbyRoleResponse } from "$lib/firebase/firestore-functions-types";
+
+  export let lobbyCode: string;
+  export let privatePlayer: PrivatePlayer | undefined;
+  export let lobby: Lobby | undefined;
+
+  let end: "Cat Win" | "Cat Lose" | "Catfish Lose" | "Catfish Win" | null = null;
+  let catfishList: string = "";
+  let catfishDisplayname: string[] = [];
+
+  function findResult(privatePlayer: PrivatePlayer, catCount: number, catfishCount: number) {
+    const { role } = privatePlayer;
+    if (role == "CAT") {
+      if (catfishCount == 0) {
+        return "Cat Win";
+      }
+      if (catCount == catfishCount) {
+        return "Cat Lose";
+      }
+      if (catCount < catfishCount) {
+        return "Cat Lose";
+      }
+    } else {
+      if (catCount == 0) {
+        return "Catfish Win";
+      }
+      if (catfishCount == catCount) {
+        return "Catfish Win";
+      }
+      if (catfishCount == 0) {
+        return "Catfish Lose";
+      }
+    }
+    return null;
+  }
+
+  onMount(async () => {
+    // function to count the number of players currently alive
+    if (lobby !== undefined) {
+      let gameInfo = await getLobbyRole({ code: lobbyCode }); // grabs return statement from function as an object
+      if ("error" in (gameInfo.data as ErrorResponse)) {
+        console.log("Error occurred!");
+      }
+      const response = gameInfo.data as LobbyRoleResponse;
+      let aliveCatCount = response.aliveCatCount; // number of cats that are currently alive
+      let catfishDisplayname = response.catfishDisplayname; // catfish's dislplay name used in the html lobbyCode
+
+      let aliveCatFishCount = catfishDisplayname.length;
+      const formatter = new Intl.ListFormat("en", { style: "long", type: "conjunction" });
+      catfishList = formatter.format(catfishDisplayname);
+      // any other player who is alive and not a cat is a catfish
+      if ($user !== null) {
+        // set conditions for the game to be declared a win or loss
+        if (privatePlayer !== undefined) {
+          end = findResult(privatePlayer, aliveCatCount, aliveCatFishCount);
+        }
+        console.log(end);
+      }
+    }
+  });
+</script>
+
+<div class="container">
+  <!-- Check the user's role and display the correct win or loss page accordingly -->
+  {#if end == "Cat Win"}
+    <div class="banner">
+      <h2>Hooray!</h2>
+      <h2>You have sniffed out the cat fish!</h2>
+    </div>
+
+    <div class="image" />
+
+    <div class="displayname">
+      {#if catfishDisplayname.length > 1}
+        <h2>{catfishList} were sentenced to 9 lives in purrison!</h2>
+      {:else}
+        <h2>{catfishList} was sentenced to 9 lives in purrison!</h2>
+      {/if}
+      {#if lobby !== undefined}
+        <button
+          on:click={async () => {
```

Comment:
you don't need async here

Reply:
I'm doing it the same way start game is doing it so wouldn't the button invoking be the same here too?

Reply:
it's also unnecessary on the start game button thanks for pointing that out